providers/amazon/tests/unit/amazon/aws/executors/aws_lambda/docker/test_app.py

```diff
@@ -0,0 +1,440 @@
+            mock_result.returncode = 1
+            mock_run.return_value = mock_result
+            mock_run.side_effect = Exception("Subprocess failed")
+
+            mock_sqs = mock_sqs_client.return_value
+
+            # Execute
+            lambda_handler(event, mock_context)
+
+            # Assert
+            mock_run.assert_called_once()
+            mock_sqs_client.assert_called_once()
+            mock_sqs.send_message.assert_called_once()
+
+            # Check that return code is 1 (failure)
+            call_args = mock_sqs.send_message.call_args
+            message_body = json.loads(call_args[1]["MessageBody"])
+            assert message_body[RETURN_CODE_KEY] == 1
+
+    def test_run_and_report_success(self, mock_sqs_client, mock_subprocess_run):
+        """Test successful execution of run_and_report."""
+        # Setup
+        command = ["airflow", "version"]
+        task_key = "test-task-key"
+
+        mock_sqs = mock_sqs_client.return_value
+
+        # Execute
+        run_and_report(command, task_key)
+
+        # Assert - Match the actual call parameters from the source code
+        mock_subprocess_run.assert_called_once_with(
+            command,
+            check=False,
+            shell=False,  # command is a list, so shell=False
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        mock_sqs_client.assert_called_once()
+        mock_sqs.send_message.assert_called_once()
+
+        # Check the message sent to SQS
+        call_args = mock_sqs.send_message.call_args
+        assert call_args[1]["QueueUrl"] == "https://sqs.us-east-1.amazonaws.com/123456789012/test-queue"
+        message_body = json.loads(call_args[1]["MessageBody"])
+        assert message_body[TASK_KEY_KEY] == task_key
+        assert message_body[RETURN_CODE_KEY] == 0
+
+    def test_run_and_report_string_command(self, mock_sqs_client):
+        """Test run_and_report with string command (shell=True)."""
+        # Setup
+        command = "airflow version"  # String command
+        task_key = "test-task-key"
+
+        with patch("airflow.providers.amazon.aws.executors.aws_lambda.docker.app.subprocess.run") as mock_run:
+            mock_result = MagicMock()
+            mock_result.returncode = 0
+            mock_result.stdout = b"Airflow version output"
+            mock_run.return_value = mock_result
+
+            # Execute
+            run_and_report(command, task_key)
+
+            # Assert - shell should be True for string commands
+            mock_run.assert_called_once_with(
+                command,
+                check=False,
+                shell=True,  # command is a string, so shell=True
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+
+    def test_run_and_report_sqs_exception(self, mock_sqs_client, mock_subprocess_run, caplog):
+        """Test run_and_report when SQS send_message raises an exception."""
+        # Setup

```

Comment:
Nit: Personally I don't think we need the `# Setup`, `# Execute`,  `# Assert` comments generally here. This is also a very AI thing to put in the code.

If you have something insightful to say at any of those times for a particular test, certainly add the context in a comment, otherwise it just adds  ~50 (!) unnecessary lines to the module.